x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestGetPipelineResponse, IngestPipeline } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { ErrorCode } from '../../common/types/error_codes';
+
+import { formatMlPipelineBody } from './create_pipeline_definitions';
+
+/**
+ * Details of a created or updated pipeline.
+ */
+export interface ModifiedPipeline {
+  id: string;
+  created?: boolean;
+  updated?: boolean;
```

Comment:
What's the `updated` property for? AFAIK, we are only creating the pipeline at the moment, correct? Or, is this intended for future use?

Reply:
These are indicator flags about processing stages (along with the ID):
- Did the creation of the inference processor succeed?
- Was it added successfully to the `{index-name}@ml-inference` pipeline?

The API tier can then decide what to do with these details.

I agree that `updated` is misleading as it refers to the _parent_ pipeline. How about changing it to `addedToParentPipeline?: boolean`?

Reply:
I think that's more clear - and especially along with wrapping two parts of the routing call into a single method ([as discussed below](https://github.com/elastic/kibana/pull/140645#discussion_r972232730))